### ISSUE TITLE
Attribute error when saving a new value

### ISSUE THREAD

**Issue by morungos** [pughlab/tracker#43]
In the webapp, we get an error on PUT when writing a field which has too long an attribute name. The length of an attribute name shouldn't be an issue, because really we should be using integer identifiers instead, but there you go. We should fix that one day. In the meantime:

```
Caused by: com.mysql.jdbc.MysqlDataTruncation: Data truncation: Data too long for column 'attribute' at row 1
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3595)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3529)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:1990)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2151)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2625)
	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:2119)
	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2415)
	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2333)
	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2318)
	at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105)
	at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105)
	at com.mysema.query.sql.dml.SQLInsertClause.execute(SQLInsertClause.java:380)
	at ca.uhnresearch.pughlab.tracker.dao.impl.StudyRepositoryImpl$16.doInSqlInsertClause(StudyRepositoryImpl.java:770)
	at org.springframework.data.jdbc.query.QueryDslJdbcTemplate$8.doInConnection(QueryDslJdbcTemplate.java:263)
```

**Comment by morungos** [comment]
Basically, let's get referential integrity to handle this for us. We actually started a bit on this in the node.js era, but it's probably more pressing now.

**Comment by morungos** [comment]
This is mostly resolved by #44, and anything left over will be handled by #46
